bindings/java/mongocrypt/build.gradle.kts

```diff
@@ -52,7 +52,7 @@ java {
 
 
 dependencies {
-    api("org.mongodb:bson:3.10.1")   // TODO: what version to depend on?
+    api("org.mongodb:bson:[3.10,4.1)")
```

Comment:
@jyemin this range should be sufficient enough for now.

Reply:
We know that 4.0 is compatible, and therefore 4.x will be compatible, so let's make the outer limit 5.0.  That way we don't need to re-publish this when we release the 4.1 driver.

Reply:
Done